test/System/UniqueResource.test.cpp

```diff
@@ -0,0 +1,107 @@
+{
+    void operator()(int)
+    {
+        ++deleteCount;
+    }
+};
+
+static_assert(!std::is_copy_constructible_v<sf::UniqueResource<int, Deleter>>);
+static_assert(!std::is_copy_assignable_v<sf::UniqueResource<int, Deleter>>);
+static_assert(std::is_nothrow_move_constructible_v<sf::UniqueResource<int, Deleter>>);
+static_assert(std::is_nothrow_move_assignable_v<sf::UniqueResource<int, Deleter>>);
+
+TEST_CASE("[System] sf::UniqueResource")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            const auto currentDeleteCount = deleteCount;
+            {
+                const sf::UniqueResource<int, Deleter> uniqueResource;
+                CHECK(uniqueResource.get() == 0);
+            }
+            CHECK(deleteCount == currentDeleteCount);
+        }
+
+        SUBCASE("Resource and deleter constructor")
+        {
+            const auto currentDeleteCount = deleteCount;
+            {
+                const sf::UniqueResource<int, Deleter> uniqueResource(42, {});
+                CHECK(uniqueResource.get() == 42);
+            }
+            CHECK(deleteCount == currentDeleteCount + 1);
+        }
+
+        SUBCASE("Move constructor")
+        {
+            const auto currentDeleteCount = deleteCount;
+            {
+                sf::UniqueResource<int, Deleter>       movedFromUniqueResource(42, {});
+                const sf::UniqueResource<int, Deleter> uniqueResource(std::move(movedFromUniqueResource));
+            }
+            CHECK(deleteCount == currentDeleteCount + 1);
+        }
+    }
+
+    SUBCASE("release()")
+    {
+        const auto currentDeleteCount = deleteCount;
+        {
+            sf::UniqueResource<int, Deleter> uniqueResource(42, {});
+            uniqueResource.release();
+        }
+        CHECK(deleteCount == currentDeleteCount);
+    }
+
+    SUBCASE("reset()")
+    {
+        const auto currentDeleteCount = deleteCount;
+        {
+            sf::UniqueResource<int, Deleter> uniqueResource(42, {});
+            uniqueResource.reset();
+            CHECK(uniqueResource.get() == 42);
+            CHECK(deleteCount == currentDeleteCount + 1);
+        }
+        CHECK(deleteCount == currentDeleteCount + 1);
+    }
+
+    SUBCASE("reset(T&&)")
+    {
+        const auto currentDeleteCount = deleteCount;
+        {
+            sf::UniqueResource<int, Deleter> uniqueResource(42, {});
+            uniqueResource.reset(123);
+            CHECK(uniqueResource.get() == 123);
+            CHECK(deleteCount == currentDeleteCount + 1);
+        }
+        CHECK(deleteCount == currentDeleteCount + 2);
+    }
+
+    SUBCASE("Operators")
+    {
+        SUBCASE("Move assignment")
+        {
+            const auto currentDeleteCount = deleteCount;
+            {
+                sf::UniqueResource<int, Deleter>       movedFromUniqueResource(42, {});
+                const sf::UniqueResource<int, Deleter> uniqueResource = std::move(movedFromUniqueResource);
```

Comment:
This is actually testing the move constructor.

Reply:
Is that because I'm using ` = ` to begin the lifetime of the object?

Reply:
Fixed 👍